musicbrainz/csv/get_genre.py

```diff
@@ -0,0 +1,62 @@
+"""
+Script for retrieving the "genre" dumps since they are not available online.
+"""
+
+import time
+import requests
+from requests.exceptions import HTTPError
+import pandas as pd
+from bs4 import BeautifulSoup
+
+URL = "https://musicbrainz.org/ws/2/genre/all"
+PARAMS = dict(fmt="json", limit="50")
+
+data = []
+max_records = dict(requests.get(url=URL, params=PARAMS, timeout=500).json())[
+    "genre-count"
+]
+
+for i in range(0, max_records):
+    PARAMS["offset"] = i
+    for j in range(0, 3):
```

Comment:
Why did you choose 3?

Reply:
If I don't give it a number for how many times I want to retry a request, then how should I deal with errors?

Reply:
I didn't say don't have a certain number of times to retry...

Walk me through in words what happens here if the request returns a 503 error.

Reply:
Ok. Let's say we request 50 records per iteration. The counter i increment by 50 each loop. 
In the nested loop, we try to make a request.get call. If it does not raise_for_status, then the nested loop breaks and continues. 
If it raise an error, it goes into the except clause. If the error is 503, then we sleep for j seconds, where j is the number of requests.get API calls we already made to the same set of 50 records, and we try again. Once we get 3 errors continuously for one set of 50 records, we raise an error.

Reply:
It looks to me that after the 3rd `503` error you will exit the loop without raising an error, no?

Reply:
Yes.

Reply:
The point of rate limiting your requests is never to actually get a 503! If you get a 503 you’re abusing the service! You need to write your code so that you do it as fast as you can but no faster.

Reply:
Right now it goes in the rate of 1/25 secs per request. If it get a 503, it waits for the number of seconds that is the number of times it reties.